hkmc2/shared/src/main/scala/hkmc2/semantics/Symbol.scala

```diff
@@ -85,15 +85,35 @@ abstract class Symbol(using State) extends Located:
   
   def asClsLike: Opt[ClassSymbol | ModuleOrObjectSymbol | PatternSymbol] =
     (asCls: Opt[ClassSymbol | ModuleOrObjectSymbol | PatternSymbol]) orElse asModOrObj orElse asPat
-  def asTpe: Opt[TypeSymbol] = asCls orElse asAls
+  def asTpe: Opt[TypeSymbol] = asCls
+    .orElse[TypeSymbol](asModOrObj)
+    .orElse[TypeSymbol](asAls)
+  def asNonModTpe: Opt[TypeSymbol] = asCls
+    .orElse[TypeSymbol](asObj)
+    .orElse[TypeSymbol](asAls)
   
   def asBlkMember: Opt[BlockMemberSymbol] = this match
     case mem: BlockMemberSymbol => S(mem)
     case mem: MemberSymbol[?] => mem.defn match
       case S(defn: TypeLikeDef) => S(defn.bsym)
       case S(defn: TermDefinition) => S(defn.sym)
       case N => N
-  
+
+  /** 
+   * Get the symbol as the "representative" symbol of a set of
+   * overloaded definitions, or the sole definition, if it is not
+   * overloaded. 
+   *
+   * We should consider the ordering terms > classes/objects/types >
+   * modules, for this purpose.
+   * */
```

Comment:
I don't know why you mangled my masterful comment by ruining its beautiful layout with an arbitrarily low line length restruction.
```suggestion
  /** Whether this definition is the "representative" definition of a set of overloaded definitions,
    * or the sole definition, if it is not overloaded.
    * We should consider the ordering terms > classes/objects/types > modules, for this purpose. */
```
FYI we consider that there's no need to wrap lines less than 100 characters. And 100+ɛ is okay.

Reply:
I have a plugin that automatically wraps the lines. 🫣

Reply:
Now changing it to 100+ɛ